### PR TITLE
add direction to mapping file and include it in packet flag

### PR DESCRIPTION
**Pull request by sensokame**
mappin.json file can now suppor direction flag (can be set to "Tx" or "Rx") which will be parsed and when applicable added to the packet flags as inbound/outboud

### CODE REVIEW

**Comment by sensokame** [comment]
@kayoub5 this is the pr to support direction in mapping file, can you check it

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 /.vs/
 /out/
+/build/**/*
```
Comment (kayoub5) [inline .gitignore:3]:
```suggestion
/build/
```

File: pcapng_exporter/mapping.cpp [hunk 2]
```diff
@@ -37,6 +37,9 @@ namespace pcapng_exporter {
 		if (when.inf_name.has_value() && when.inf_name != target.inf_name) {
 			return false;
 		}
+		if (when.direction.has_value() && when.direction!= target.direction){
```
Comment (kayoub5) [inline pcapng_exporter/mapping.cpp:40]:
```suggestion
		if (when.direction.has_value() && when.direction != target.direction){
```

File: pcapng_exporter/include/pcapng_exporter/mapping.hpp [hunk 3]
```diff
@@ -16,6 +16,7 @@ namespace pcapng_exporter {
 	struct channel_info {
 		std::optional<std::uint32_t> chl_id = std::nullopt;
 		std::optional<std::string> inf_name = std::nullopt;
+		std::optional<std::string> direction = std::nullopt;
```
Comment (kayoub5) [inline pcapng_exporter/include/pcapng_exporter/mapping.hpp:19]: * Document valid values for direction
* Would an enum be better than a string here? for performance

File: pcapng_exporter/mapping.cpp [hunk 4]
```diff
@@ -47,13 +50,15 @@ namespace pcapng_exporter {
 		target.inf_name = packet_interface.name
 			? std::optional(std::string(packet_interface.name))
 			: std::nullopt;
+		target.direction = std::nullopt;
```
Comment (kayoub5) [inline pcapng_exporter/mapping.cpp:53]: TECMP provides Tx / Rx, pass it through

File: pcapng_exporter/pcapng_exporter.cpp [hunk 5]
```diff
@@ -66,7 +69,7 @@ namespace pcapng_exporter {
 	void PcapngExporter::write_packet(
 		uint32_t channel_id,
 		const light_packet_interface packet_interface,
-		const light_packet_header packet_header,
+		light_packet_header packet_header,
```
Comment (kayoub5) [inline pcapng_exporter/pcapng_exporter.cpp:72]: Revert this, `write_packet` is not allowed to change `packet_header`

File: tests/mapping.json [hunk 6]
```diff
@@ -6,15 +6,17 @@
           "chl_id": "0xca"
         },
         "change": {
-          "inf_name": "can0"
+          "inf_name": "can0",
+          "direction": "Tx"
```
Comment (kayoub5) [inline tests/mapping.json:10]:
```suggestion
          "pkt_dir": "Tx"
```

File: tests/mapping.json [hunk 7]
```diff
@@ -6,15 +6,17 @@
...
         }
       },
       {
         "when": {
           "chl_id": 3
         },
         "change": {
-          "inf_name": "can_three"
+          "inf_name": "can_three",
+          "direction": "Rx"
```
Comment (kayoub5) [inline tests/mapping.json:19]:
```suggestion
          "pkt_dir": "Rx"
```

**Review by kayoub5 — CHANGES_REQUESTED** [review]

File: pcapng_exporter/include/pcapng_exporter/mapping.hpp (hunk 3 above)
Comment (sensokame) [reply pcapng_exporter/include/pcapng_exporter/mapping.hpp]: switched to an enum and documented possible string values in json file

File: pcapng_exporter/mapping.cpp (hunk 4 above)
Comment (sensokame) [reply pcapng_exporter/mapping.cpp]: passed it on as a value in the header flag and then affected it to the target

**Comment by sensokame** [comment]
@kayoub5 can you review again?

**Review by kayoub5 — APPROVED** [review]

**Comment by kayoub5** [comment]
@sensokame could you rebase the PR?

**Comment by sensokame** [comment]
@kayoub5 done

**Pull request merged**